File: sample/src/main/java/com/sha/formvalidatorsample/validator/NumberOneCustomValidator.java
package com.sha.formvalidatorsample.validator;

import android.content.Context;
import android.widget.EditText;

import com.sha.formvalidator.validator.CustomValidator;
import com.sha.formvalidatorsample.R;

public class NumberOneCustomValidator extends CustomValidator {

    public NumberOneCustomValidator(String errorMessage) {
        super(errorMessage);
    }

    @Override
    public String customValidationType(Context context) {
        return context.getString(R.string.custom_validator_number_one);
    }

    @Override
    public boolean isValid(EditText et) {
        return et.getText().toString().equals("1");
    }

}

File: sample/src/main/java/com/sha/formvalidatorsample/validator/NumberOneCustomValidator.kt
package com.sha.formvalidatorsample.validator

import android.content.Context
import android.widget.TextView
import com.sha.formvalidator.textview.validator.CustomValidator
import com.sha.formvalidatorsample.R

class NumberOneCustomValidator(errorMessage: String) : CustomValidator(errorMessage) {

    override fun customValidationType(context: Context): String {
        return context.getString(R.string.custom_validator_number_one)
    }

    override fun isValid(tv: TextView): Boolean {
        return tv.text.toString() == "1"
    }

}
